### PR TITLE
Feat/course routes2

### PR DESCRIPTION
**Pull request by nugentmichael1**

### CODE REVIEW

File: src/main/java/com/revlearn/team1/dto/CourseDTO.java [hunk 1]
```diff
@@ -4,17 +4,18 @@
...
         User institution,
         LocalDate startDate,
         LocalDate endDate,
-        AttendanceMethod type,
+        AttendanceMethod attendanceMethod,
         String name,
         String description,
-        Set<User> students,
-        Set<User> educators
+        List<User> students,
```
Comment (treycrossley) [inline src/main/java/com/revlearn/team1/dto/CourseDTO.java:17]: why list over set? Wouldn't set make more sense for this

File: src/main/java/com/revlearn/team1/model/Course.java [hunk 2]
```diff
@@ -8,57 +12,72 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
     @ManyToMany
     @JoinTable(
             name = "course_educators",
             joinColumns = @JoinColumn(name = "course_id"),
-            inverseJoinColumns = @JoinColumn(name = "educator_id")
+            inverseJoinColumns = @JoinColumn(name = "educator_id"),
+            uniqueConstraints = @UniqueConstraint(columnNames = {"course_id", "educator_id"})  // Ensure uniqueness
```
Comment (treycrossley) [inline src/main/java/com/revlearn/team1/model/Course.java:33]: you're enforcing unique constraints but use a list over a set. Might be a good idea to change that

File: src/main/java/com/revlearn/team1/model/Course.java [hunk 3]
```diff
@@ -8,57 +12,72 @@
...
     )
-    private Set<User> educators = new HashSet<>();
+    @JsonIgnore
+    private List<User> educators = new ArrayList<>();
```
Comment (treycrossley) [inline src/main/java/com/revlearn/team1/model/Course.java:36]: same here with List over Set

File: src/main/java/com/revlearn/team1/model/Course.java [hunk 4]
```diff
@@ -8,57 +12,72 @@
...
+
     @ManyToMany
     @JoinTable(
             name = "course_students",
             joinColumns = @JoinColumn(name = "course_id"),
-            inverseJoinColumns = @JoinColumn(name = "student_id")
+            inverseJoinColumns = @JoinColumn(name = "student_id"),
+            uniqueConstraints = @UniqueConstraint(columnNames = {"course_id", "student_id"})  // Ensure uniqueness
```
Comment (treycrossley) [inline src/main/java/com/revlearn/team1/model/Course.java:43]: set

**Review by treycrossley — APPROVED** [review]
Looks good! The set comment isn't really all that important so we can ignore it for now!

**Comment by nugentmichael1** [comment]
There was something weird about the Set collection that made the @ManyToMany relationships fail.  Database retrievals would not populate the mapped entities despite clear evidence they were connected via the PostgreSQL join table.  
I might try again with hashCode() and equals() implementations.

**Pull request merged**